handler/src/main/java/io/netty/handler/ssl/PemPrivateKey.java

```diff
@@ -0,0 +1,211 @@
+                SslUtils.zerooutAndRelease(base64);
+            }
+        } finally {
+            SslUtils.zerooutAndRelease(encoded);
+        }
+    }
+
+    /**
+     * Creates a {@link PemPrivateKey} from raw {@code byte[]}.
+     *
+     * ATTENTION: It's assumed that the given argument is a PEM/PKCS#8 encoded value.
+     * No input validation is performed to validate it.
+     */
+    public static PemPrivateKey valueOf(byte[] key) {
+        return valueOf(Unpooled.wrappedBuffer(key));
+    }
+
+    /**
+     * Creates a {@link PemPrivateKey} from raw {@code ByteBuf}.
+     *
+     * ATTENTION: It's assumed that the given argument is a PEM/PKCS#8 encoded value.
+     * No input validation is performed to validate it.
+     */
+    public static PemPrivateKey valueOf(ByteBuf key) {
+        return new PemPrivateKey(key);
+    }
+
+    private final ByteBuf content;
+
+    private PemPrivateKey(ByteBuf content) {
+        this.content = ObjectUtil.checkNotNull(content, "content");
+    }
+
+    @Override
+    public boolean isSensitive() {
+        return true;
+    }
+
+    @Override
+    public ByteBuf content() {
+        int count = refCnt();
+        if (count <= 0) {
+            throw new IllegalReferenceCountException(count);
+        }
+
+        return content;
+    }
+
+    @Override
+    public PemPrivateKey copy() {
+        return replace(content.copy());
+    }
+
+    @Override
+    public PemPrivateKey duplicate() {
+        return replace(content.duplicate());
+    }
+
+    @Override
+    public PemPrivateKey retainedDuplicate() {
+        return replace(content.retainedDuplicate());
+    }
+
+    @Override
+    public PemPrivateKey replace(ByteBuf content) {
+        return new PemPrivateKey(content);
+    }
+
+    @Override
+    public PemPrivateKey touch() {
+        content.touch();
+        return this;
+    }
+
+    @Override
+    public PemPrivateKey touch(Object hint) {
+        content.touch(hint);
+        return this;
+    }
+
+    @Override
+    public PemPrivateKey retain() {
+        return (PemPrivateKey) super.retain();
+    }
+
+    @Override
+    public PemPrivateKey retain(int increment) {
+        return (PemPrivateKey) super.retain(increment);
+    }
+
+    @Override
+    protected void deallocate() {
+        // Private Keys are sensitive. We need to zero the bytes
+        // before we're releasing the underlying ByteBuf
+        SslUtils.zerooutAndRelease(content);
+    }
+
+    @Override
+    public byte[] getEncoded() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getAlgorithm() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getFormat() {
+        return PKCS8_FORMAT;
+    }
+
+    /**
+     * @see Destroyable#destroy()
+     */
+    public void destroy() {
+        release(refCnt());
+    }
+
+    /**
+     * @see Destroyable#isDestroyed()
+     */
+    public boolean isDestroyed() {
```

Comment:
I'm a bit on the fence. Java's PrivateKey interface extends Destroyable. No idea when that was introduced. I assume JDK8 because its got `default` impls. straight on the interface. Yesterday's build errors were because of that.